bootstrap.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable import/no-dynamic-require,global-require */
+const fs = require('fs');
+const path = require('path');
+const spawn = require('child_process').spawnSync;
+
+const type = process.env.type || 'upgrade'; // yarn update types
+const logTask = msg => console.log(`👍  ${msg}`);
+
+const installPackage = (at) => {
+  const result = spawn('yarn', [type], { stdio: 'inherit', cwd: at });
+  if (result.error) {
+    console.log(result.error);
+    process.exit(1);
+  }
+  logTask(`installed ${at}\n`);
+};
+
+const packages = fs.readdirSync('packages').reduce((pkgs, pkg) => {
+  const packagePath = path.join(process.cwd(), 'packages', pkg);
+  const packageJSON = path.join(packagePath, 'package.json');
+  try {
+    if (fs.statSync(packagePath).isDirectory() && fs.statSync(packageJSON).isFile()) {
+      pkgs.push({ path: packagePath, name: require(packageJSON).name });
+    }
+  } catch (e) { return pkgs; }
+  return pkgs;
+}, []);
+
+console.log(`\n🔥  Bootstrapping\n`);
+
+// Install the root package.json first so that we can use shelljs.
+installPackage(process.cwd());
+const shell = require('shelljs');
+
+// Install all of the monorepo packages.
+packages.forEach(pkg => installPackage(pkg.path));
+
+// Symlink monorepo package dependencies to local packages.
+packages.forEach((pkg) => {
```

Comment:
All of the  kyt-specific dependencies in the monorepo should be in sync with the latest published versions. Symlinking makes it easy to do local development on a specific kyt package and its respective kyt dependencies (eg changes to kyt-core and kyt-utils). It's true that this makes the assumption that things will stay in sync, but the symlinks are also easy to override after bootstrap if there is an issue or you want to test against a specific version. Let me know if I can clarify more.